src/main/scala/io/iohk/ethereum/network/PeerActor.scala

```diff
@@ -295,29 +297,51 @@ class PeerActor(
 
   class HandshakedHandler(rlpxConnection: RLPxConnection, initialStatus: msg.Status) {
 
+    var currentPeerMaxBlock: BigInt = 0
+
     def receive: Receive =
       handleSubscriptions orElse handleTerminated(rlpxConnection) orElse
       handlePeerChainCheck(rlpxConnection) orElse handlePingMsg(rlpxConnection) orElse
       handleBlockFastDownload(rlpxConnection) orElse {
       case RLPxConnectionHandler.MessageReceived(message) =>
         log.debug("Received message: {}", message)
+        updateMaxBlock(message)
```

Comment:
can we also update max block in `waitingForChainForkCheck`? we receive block headers there